src/aes_xts_plug.c

```diff
@@ -63,3 +63,46 @@ void AES_XTS_decrypt(const unsigned char *double_key, unsigned char *out,
 	}
 }
 
+void AES_XTS_decrypt_custom_tweak(const unsigned char *double_key, unsigned
+		char *tweak, unsigned char *out, const unsigned char *data,
+		unsigned len, int bits) {
+	unsigned char buf[16];
+	int i, j, cnt;
+	AES_KEY key1, key2;
+	if (bits == 256) {
+		AES_set_decrypt_key(double_key, 256, &key1);
+		AES_set_encrypt_key(&double_key[32], 256, &key2);
+	} else if (bits == 128) {
+		AES_set_decrypt_key(double_key, 128, &key1);
+		AES_set_encrypt_key(&double_key[16], 128, &key2);
+	}
+	// else 192 bits???
```

Comment:
This would obviously be more generic
```diff
-	if (bits == 256) {
-		AES_set_decrypt_key(double_key, 256, &key1);
-		AES_set_encrypt_key(&double_key[32], 256, &key2);
-	} else if (bits == 128) {
-		AES_set_decrypt_key(double_key, 128, &key1);
-		AES_set_encrypt_key(&double_key[16], 128, &key2);
-	}
-	// else 192 bits???
+	AES_set_decrypt_key(double_key, bits, &key1);
+	AES_set_encrypt_key(&double_key[bits / 8], bits, &key2);
```